src/main/java/com/google/cloud/healthcare/etl/runner/dicomtofhir/DicomToFhirStreamingRunner.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.*;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The entry point of the pipeline. It will be triggered upon receiving a PubSub message from the given subscription. From
+ * the message, it will extract the webpath  of a DICOM instance and collect it's metadata from DICOM IO. With the
+ * response, it will map the study metadata to a FHIR ImagingStudyResource and upload it to the given FHIR store.
+ *
+ * The errors for each component are handled separately, e.g. you can specify file paths for each
+ * of the stage (read - DICOM IO, mapping, write - FHIR IO). Right now the shard is set to 1, if
+ * you are seeing issues with regard to writing to GCS, feel free to bump it up to a reasonable
+ * value.
+ *
+ */
+public class DicomToFhirStreamingRunner {
+    private static final Logger LOG = LoggerFactory.getLogger(DicomToFhirStreamingRunner.class);
+    private static Duration ERROR_LOG_WINDOW_SIZE = Duration.standardSeconds(5);
+
+    /**
+    * A DoFn to collect the webpath of the DICOM instance from the PubSub Message.
+     */
+    public static class ExtractWebpathFromPubsub extends DoFn<PubsubMessage, String> {
+        @ProcessElement
+        public void processElement(DoFn<PubsubMessage, String>.ProcessContext context) throws UnsupportedEncodingException {
+            PubsubMessage msg = context.element();
+            byte[] msgPayload = msg.getPayload();
+            String webpath = new String(msgPayload, StandardCharsets.UTF_8);
+            context.output(webpath);
+            DicomToFhirStreamingRunner.LOG.info("Extracted webpath");
+        }
+    }
+
+    /**
+     * A DoFn that will take the response of the Study Metadata Read call from the DICOM API and reformat it into an
+     * input to be consumed by the mapping library.
+     */
+    public static class ReformatMappingFnInputString extends DoFn<String, String> {
+        @ProcessElement
+        public void processElement(DoFn<String, String>.ProcessContext context) {
+            String dicomMetadataResponse = context.element();
+            Gson gson = new Gson();
+            JsonArray jsonArray = gson.fromJson(dicomMetadataResponse, JsonArray.class);
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.add("study", jsonArray);
+            context.output(gson.toJson(jsonObject));
+            DicomToFhirStreamingRunner.LOG.info("Sending instance to be mapped");
+        }
+    }
+
+    /**
+     * A DoFn that will take the response of the mapping library and wrap it into a FHIR bundle to be written to the
+     * FHIR store.
+     * TODO(b/174594428): Add a unique ID for each ImagingStudy FHIR resource to be uploaded to prevent creation of multiple FHIR resources for each ImagingStudy.
```

Comment:
how about "to prevent creation of one FHIR resource for every DICOM instance in an ImagingStudy"?